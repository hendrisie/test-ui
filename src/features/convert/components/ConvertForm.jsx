import React from 'react';
import { Field } from 'redux-form';
import FormTextField from '../../../components/FormTextField';
import {
  validate,
  // parser,
} from '../formhelper';

const ConvertForm = props => {
  const { handleSubmit, classes } = props;
  return (
    <form onSubmit={handleSubmit}>
      <div>
        <Field
          classes={classes}
          name="total"
          component={FormTextField}
          label="Currency Value"
          placeholder="Please input value"
          validate={[validate.required, validate.validCurrency]}
          // parse={parser.currencyParser}
        />
      </div>
    </form>
  );
};

export default ConvertForm;
