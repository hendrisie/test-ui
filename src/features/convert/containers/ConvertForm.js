import { reduxForm } from 'redux-form';
import component from '../components/ConvertForm';
import { submitTotal } from '../actions';

const submitForm = (values, dispatch) => {
  return dispatch(submitTotal(values.total));
};

// const validate = values => {
//   const errors = {};
//   const requiredFields = ['total'];
//   requiredFields.forEach(field => {
//     if (!values[field]) {
//       errors[field] = 'Required';
//     }
//   });
//   const parsedFields = ['total'];
//   const regex = /^(Rp)?(\s+)?(\d{1,3}(\.\d{3})*|(\d+))(\,\d{0,2})?$/;
//   parsedFields.forEach( field => {
//     if (!regex.test(values[field])) {
//       errors[field] = 'Wrong Format';
//     }
//   });
//   return errors;
// };

const ConvertForm = reduxForm({
  form: 'ConvertForm', // a unique identifier for this form
  onSubmit: submitForm
})(component);

export default ConvertForm;
