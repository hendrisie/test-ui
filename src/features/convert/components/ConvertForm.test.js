import React from 'react';
import { shallow } from 'enzyme';
import ConvertForm from './ConvertForm';
describe('ConvertForm', () => {
  it('whould render correctly', () => {
    const component = shallow(<ConvertForm />);
    expect(component).toMatchSnapshot();
  });
});