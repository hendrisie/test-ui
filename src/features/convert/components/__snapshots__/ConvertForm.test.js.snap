// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConvertForm whould render correctly 1`] = `ShallowWrapper {}`;
